manager/dispatcher/dispatcher.go

```diff
@@ -572,20 +573,45 @@ func (d *Dispatcher) Tasks(r *api.TasksRequest, stream api.Dispatcher_TasksServe
 			return err
 		}
 
-		select {
-		case event := <-nodeTasks:
-			switch v := event.(type) {
-			case state.EventCreateTask:
-				tasksMap[v.Task.ID] = v.Task
-			case state.EventUpdateTask:
-				tasksMap[v.Task.ID] = v.Task
-			case state.EventDeleteTask:
-				delete(tasksMap, v.Task.ID)
+		// bursty events should be processed in batches and sent out snapshot
+		const eventBatchLimit = 200
+		const eventPausedGap = 50 * time.Millisecond
+		var eventCnt int
+		// eventPaused is true when there are events
+		// but next event has not arrived within eventPausedGap
+		eventPaused := false
+
+		for eventCnt < eventBatchLimit && !eventPaused {
+			select {
+			case event := <-nodeTasks:
```

Comment:
@aluzzardi It doesn't solve the problem. Task status update from agent also triggers commit to store. We don't want to send update on that. 

If you want to listen on commit event, then query store to find task difference. It would work. But every dispatcher reacts to this event, it'd result in a spike load on store. It wouldn't work on large cluster.